### PR TITLE
piecestore: pipeline chunks with orders

### PR DESCRIPTION
**Pull request by jtolds**
What: send orders and data with those orders together

Why: less messages

Please describe the tests: all existing tests should cover this
 - Test 1:
 - Test 2:
 
Please describe the performance impact: should reduce message overhead

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: storagenode/piecestore/endpoint.go [hunk 1]
```diff
@@ -227,17 +227,14 @@ func (endpoint *Endpoint) Upload(stream pb.Piecestore_UploadServer) (err error)
 			return ErrProtocol.New("expected a message") // TODO: report grpc status bad message
 		}
 
-		switch {
-		default:
-			return ErrProtocol.New("message didn't contain any of order, chunk or done") // TODO: report grpc status bad message
-
-		case message.Order != nil:
+		if message.Order != nil {
```
Comment (egonelbre) [inline storagenode/piecestore/endpoint.go]: Let's check that at least one of them is present.

**Comment by egonelbre** [comment]
Also wondering, doesn't grpc have buffering etc. to handle this?

**Comment by egonelbre** [comment]
@mniewrzal yeah, but we can implement that later since it won't be backwards incompatible.

**Comment by mniewrzal** [comment]
yes, just a note

**Comment by mniewrzal** [comment]
https://storjlabs.atlassian.net/browse/V3-2102

**Pull request merged**